packages/devtools/lib/src/timeline/frame_flame_chart.dart

```diff
@@ -407,3 +430,142 @@ class FlameChartItem {
         colorToCss(selected ? selectedTextColor : defaultTextColor);
   }
 }
+
+class TimelineGrid extends CoreElement {
+  TimelineGrid(this._frameDurationMs, this.flameChartWidth)
+      : super('div', classes: 'flame-chart-grid') {
+    flex();
+    layoutHorizontal();
+    _drawGrid(baseGridInterval);
+  }
+
+  static const baseGridInterval = 150;
+
+  final num _frameDurationMs;
+
+  num zoomLevel = 1;
+
+  num flameChartWidth;
+
+  num get flameChartWidthWithInsets => flameChartWidth + 2 * flameChartInset;
+
+  num interval = baseGridInterval;
+
+  final List<TimelineGridItem> _gridItems = [];
+
+  void _drawGrid(num interval) {
+    // Draw the first grid item since it will have a different width than the
+    // rest.
+    final gridItem = TimelineGridItem(0, flameChartInset, 0);
+    _gridItems.add(gridItem);
+    add(gridItem);
+
+    num left = flameChartInset;
+
+    while (left + interval < flameChartWidthWithInsets) {
+      final timestamp = getTimestampForPosition(left + interval);
+      final gridItem = TimelineGridItem(left, interval, timestamp);
+
+      _gridItems.add(gridItem);
+      add(gridItem);
+
+      left += interval;
+    }
+  }
+
+  void _clear() {
+    clear();
```

Comment:
having _clear call a public clear method seems a bit inverted. 
Make 
clear() private with a different name.

Reply:
I can change it to 
```
_clearGrid() {
  clear();
  _gridItems.clear();
}
```

Reply:
Actually i don't need this helper method. I'll just call `clear()` and `_gridItems.clear()` directly where I need them. This helper was left over from when I was calling this code from two different places.

Reply:
sg. 
Still a little bit of a code smell that a public method called clear() doesn't actually clear everything. You might want to remove that method and make it private unless I'm missing something/

Reply:
`clear()` is from the elements.dart class:
`void clear() => element.children.clear();`
This is how we clear an element's children throughout the code.